test/suites/ipv6/dualstack.robot

```diff
@@ -0,0 +1,178 @@
+*** Settings ***
+Documentation       Tests related to MicroShift running in a dual stack ipv4+6 host
+
+Resource            ../../resources/common.resource
+Resource            ../../resources/oc.resource
+Resource            ../../resources/ostree-health.resource
+Resource            ../../resources/microshift-network.resource
+Resource            ../../resources/microshift-config.resource
+Library             ../../resources/libipv6.py
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           ipv6    network
+
+
+*** Variables ***
+${USHIFT_HOST_IP1}      ${EMPTY}
+${USHIFT_HOST_IP2}      ${EMPTY}
+${HOSTNAME}             hello-microshift.dualstack.cluster.local
+
+
+*** Test Cases ***
+Verify New Pod Works With IPv6
+    [Documentation]    Verify IPv6 services are routable.
+    [Setup]    Run Keywords
+    ...    Save Default MicroShift Config
+    ...    Migrate To Dual Stack
+    ...    Create Hello MicroShift Pod
+    ...    Expose Hello MicroShift Service Via Route IPv6
+    ...    Restart Router
+
+    ${pod_ip}=    Run With Kubeconfig
+    ...    oc get pod hello-microshift -n ${NAMESPACE} -o jsonpath='{.status.podIPs[0].ip}'
+    Must Not Be Ipv6    ${pod_ip}
```

Comment:
How do we know that 1st IP is ipv4? Does this just happens with dual stack?

Reply:
The cluster started as IPv4 only. Then the tests enable dual stack and restart MicroShift. All of the pods are getting the IP addresses in the same order than defined (and you can not swap the primary IP family in between restarts).